javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -160,7 +140,58 @@ public static ResettableClassFileTransformer installBytebuddyAgent(
 
     agentBuilder = customizeByteBuddyAgent(agentBuilder);
     log.debug("Installed {} instrumenter(s)", numInstrumenters);
-    return agentBuilder.installOn(inst);
+    ResettableClassFileTransformer resettableClassFileTransformer = agentBuilder.installOn(inst);
+    installComponentsAfterByteBuddy(componentInstallers);
+    return resettableClassFileTransformer;
+  }
+
+  private static void installComponentsBeforeByteBuddy(
+      Iterable<ComponentInstaller> componentInstallers) {
+    ClassLoader savedContextClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(AgentInstaller.class.getClassLoader());
+      for (ComponentInstaller componentInstaller : componentInstallers) {
+        componentInstaller.beforeByteBuddyAgent();
+      }
+    } finally {
+      Thread.currentThread().setContextClassLoader(savedContextClassLoader);
+    }
```

Comment:
I think setting the context class loader would make more sense (and be more self-explanatory) up in AgentInitializer.startAgent(), e.g.

```
        ClassLoader agentClassLoader = createAgentClassLoader("inst", bootstrapUrl);
        Class<?> agentInstallerClass =
            agentClassLoader.loadClass("io.opentelemetry.javaagent.tooling.AgentInstaller");
        Method agentInstallerMethod =
            agentInstallerClass.getMethod("installBytebuddyAgent", Instrumentation.class);
        ClassLoader savedContextClassLoader = Thread.currentThread().getContextClassLoader();
        try {
          Thread.currentThread().setContextClassLoader(agentClassLoader);
          agentInstallerMethod.invoke(null, inst);
        } finally {
          Thread.currentThread().setContextClassLoader(savedContextClassLoader);
        }
```